gprpc-swift tests stuck (docker)

<!-- Thanks for helping grpc-swift! Before you submit your issue, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### New Issue Checklist

- [x] Updated grpc-swift to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/grpc/grpc-swift/blob/master/CONTRIBUTING.md)
- [x] I searched for [existing GitHub issues](https://github.com/grpc/grpc-swift/issues)(especially, there are many [resolved issues related to SSL](https://github.com/grpc/grpc-swift/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+ssl))

### Issue Description

as part of swift-nio-release tools, I run the grpc-swift tests in various configurations. The docker run has been stuck for a loooong time. grpc commit is 283754e80bb30a630e5e06508af654bc1d68fd7d . 

It seems like the promised allocated in  `file="/Users/johannes/.Trash/.nio-release-tools_qP5z8N/grpc-swift/Tests/SwiftGRPCNIOTests/ServerThrowingTests.swift", line=152` is not being fulfilled.

```
(lldb) bt
* thread #1, name = 'SwiftGRPCPackag', stop reason = signal SIGSTOP
  * frame #0: 0x00007f5a2ff41404 libpthread.so.0`__pthread_cond_wait + 196
    frame #1: 0x000055b2b10d32fa SwiftGRPCPackageTests.xctest`ConditionLock.lock(wantedValue=1, self=<unavailable>) at lock.swift:143:23
    frame #2: 0x000055b2b1013454 SwiftGRPCPackageTests.xctest`EventLoopFuture.wait(file="/Users/johannes/.Trash/.nio-release-tools_qP5z8N/grpc-swift/Tests/SwiftGRPCNIOTests/ServerThrowingTests.swift", line=152, self=<unavailable>) at EventLoopFuture.swift:873:14
    frame #3: 0x000055b2b13bba4c SwiftGRPCPackageTests.xctest`implicit closure #1 in ServerThrowingTests.testBidirectionalStreaming() at <compiler-generated>:0
    frame #4: 0x000055b2b13bbac6 SwiftGRPCPackageTests.xctest`partial apply for implicit closure #1 in ServerThrowingTests.testBidirectionalStreaming() at <compiler-generated>:0
    frame #5: 0x000055b2b0f66d56 SwiftGRPCPackageTests.xctest`thunk for @callee_guaranteed () -> (@error @owned Error) at <compiler-generated>:0
    frame #6: 0x000055b2b13bbb0b SwiftGRPCPackageTests.xctest`thunk for @callee_guaranteed () -> (@error @owned Error)partial apply at <compiler-generated>:0
    frame #7: 0x00007f5a30240d38 libXCTest.so`closure #1 () -> XCTest.(_XCTAssertionResult in _F1DE58039960C6E2C9E25CD38F016B36) in XCTest.XCTAssertNoThrow<A>(_: @autoclosure () throws -> A, _: @autoclosure () -> Swift.String, file: Swift.StaticString, line: Swift.UInt) -> () + 56
    frame #8: 0x00007f5a30240a58 libXCTest.so`XCTest.XCTAssertNoThrow<A>(_: @autoclosure () throws -> A, _: @autoclosure () -> Swift.String, file: Swift.StaticString, line: Swift.UInt) -> () + 40
    frame #9: 0x000055b2b13bb561 SwiftGRPCPackageTests.xctest`ServerThrowingTests.testBidirectionalStreaming(self=<unavailable>) at ServerThrowingTests.swift:152:5
    frame #10: 0x000055b2b13c1ec9 SwiftGRPCPackageTests.xctest`partial apply for ServerThrowingTests.testBidirectionalStreaming() at <compiler-generated>:0
    frame #11: 0x000055b2b0f66f86 SwiftGRPCPackageTests.xctest`thunk for @escaping @callee_guaranteed () -> (@error @owned Error) at <compiler-generated>:0
    frame #12: 0x000055b2b13c5dab SwiftGRPCPackageTests.xctest`thunk for @escaping @callee_guaranteed () -> (@error @owned Error)partial apply at <compiler-generated>:0
    frame #13: 0x00007f5a30231951 libXCTest.so`partial apply forwarder for reabstraction thunk helper from @escaping @callee_guaranteed () -> (@out (), @error @owned Swift.Error) to @escaping @callee_guaranteed () -> (@error @owned Swift.Error) + 17
    frame #14: 0x00007f5a302316e8 libXCTest.so`partial apply forwarder for closure #1 (XCTest.XCTestCase) throws -> () in XCTest.(test in _3BE257A46ADB477C7BF2D39968B39F9D)<A where A: XCTest.XCTestCase>((A) -> () throws -> ()) -> (XCTest.XCTestCase) throws -> () + 88
    frame #15: 0x00007f5a30231654 libXCTest.so`partial apply forwarder for reabstraction thunk helper from @escaping @callee_guaranteed (@guaranteed XCTest.XCTestCase) -> (@error @owned Swift.Error) to @escaping @callee_guaranteed (@in_guaranteed XCTest.XCTestCase) -> (@out (), @error @owned Swift.Error) + 20
    frame #16: 0x00007f5a302319b9 libXCTest.so`reabstraction thunk helper from @escaping @callee_guaranteed (@guaranteed XCTest.XCTestCase) -> (@error @owned Swift.Error) to @escaping @callee_guaranteed (@in_guaranteed XCTest.XCTestCase) -> (@out (), @error @owned Swift.Error)partial apply forwarder with unmangled suffix ".37" + 9
    frame #17: 0x00007f5a3021e257 libXCTest.so`partial apply forwarder for reabstraction thunk helper from @escaping @callee_guaranteed (@in_guaranteed XCTest.XCTestCase) -> (@out (), @error @owned Swift.Error) to @escaping @callee_guaranteed (@guaranteed XCTest.XCTestCase) -> (@error @owned Swift.Error) + 39
    frame #18: 0x00007f5a3022f76d libXCTest.so`XCTest.XCTestCase.invokeTest() -> () + 77
    frame #19: 0x00007f5a3022f34c libXCTest.so`XCTest.XCTestCase.perform(XCTest.XCTestRun) -> () + 172
    frame #20: 0x00007f5a302353f2 libXCTest.so`XCTest.XCTest.run() -> () + 194
    frame #21: 0x00007f5a30231dfe libXCTest.so`XCTest.XCTestSuite.perform(XCTest.XCTestRun) -> () + 254
    frame #22: 0x00007f5a302353f2 libXCTest.so`XCTest.XCTest.run() -> () + 194
    frame #23: 0x00007f5a30231dfe libXCTest.so`XCTest.XCTestSuite.perform(XCTest.XCTestRun) -> () + 254
    frame #24: 0x00007f5a302353f2 libXCTest.so`XCTest.XCTest.run() -> () + 194
    frame #25: 0x00007f5a30231e20 libXCTest.so`XCTest.XCTestSuite.perform(XCTest.XCTestRun) -> () + 288
    frame #26: 0x00007f5a302353f2 libXCTest.so`XCTest.XCTest.run() -> () + 194
    frame #27: 0x00007f5a3022d9d4 libXCTest.so`XCTest.XCTMain(Swift.Array<(testCaseClass: XCTest.XCTestCase.Type, allTests: Swift.Array<(Swift.String, (XCTest.XCTestCase) throws -> ())>)>) -> Swift.Never + 1652
    frame #28: 0x000055b2b13cc21d SwiftGRPCPackageTests.xctest`main at LinuxMain.swift:10:1
    frame #29: 0x00007f5a2d4f9f45 libc.so.6`__libc_start_main + 245
    frame #30: 0x000055b2b0c95c04 SwiftGRPCPackageTests.xctest`_start + 41
```

package.resolved:

```
{
  "object": {
    "pins": [
      {
        "package": "Commander",
        "repositoryURL": "https://github.com/kylef/Commander.git",
        "state": {
          "branch": null,
          "revision": "e5b50ad7b2e91eeb828393e89b03577b16be7db9",
          "version": "0.8.0"
        }
      },
      {
        "package": "Spectre",
        "repositoryURL": "https://github.com/kylef/Spectre.git",
        "state": {
          "branch": null,
          "revision": "f14ff47f45642aa5703900980b014c2e9394b6e5",
          "version": "0.9.0"
        }
      },
      {
        "package": "swift-nio-ssl-support",
        "repositoryURL": "https://github.com/apple/swift-nio-ssl-support.git",
        "state": {
          "branch": null,
          "revision": "c02eec4e0e6d351cd092938cf44195a8e669f555",
          "version": "1.0.0"
        }
      },
      {
        "package": "swift-nio-zlib-support",
        "repositoryURL": "https://github.com/apple/swift-nio-zlib-support.git",
        "state": {
          "branch": null,
          "revision": "37760e9a52030bb9011972c5213c3350fa9d41fd",
          "version": "1.0.0"
        }
      },
      {
        "package": "SwiftProtobuf",
        "repositoryURL": "https://github.com/apple/swift-protobuf.git",
        "state": {
          "branch": null,
          "revision": "6520fb185db88c0774a929acea1f7d5981a30d3a",
          "version": "1.4.0"
        }
      }
    ]
  },
  "version": 1
}
```

NIO repositories manually pinned to

```
swift-nio: e32a436466b01d4d45555356eb5bbdcdf14b4526
swift-nio-http2: 08db9d8cd4ec5dcb36787f1f7341418cd8c4fa97
swift-nio-ssl: 4f9d64ddd04cd1fe97d74dad7b5a77b882d8bf5d
```

this is Swift 5.0, on docker (ubuntu 14.04)

ok, I can repro this 100% with the regular `swift:5.0` docker image. Here's the full trace:

```
johannes:~/.Trash/.nio-release-tools_WPakBo
$ docker run -it --rm  -v "$PWD:$PWD" -w "$PWD" swift:5.0 bash
root@71109ea3bb0c:/Users/johannes/.Trash/.nio-release-tools_WPakBo# cd grpc-swift/
root@71109ea3bb0c:/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift# swift test
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/JSONScanner.swift:192:15: warning: 'withUnsafeMutableBytes' is deprecated: use `withUnsafeMutableBytes<R>(_: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R` instead
    try value.withUnsafeMutableBytes {
              ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/JSONScanner.swift:850:25: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
        let n = try raw.withUnsafeBytes {
                        ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/JSONScanner.swift:909:25: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
        let n = try raw.withUnsafeBytes {
                        ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/JSONScanner.swift:973:27: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
          let n = try raw.withUnsafeBytes {
                          ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/JSONScanner.swift:1044:27: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
          let n = try raw.withUnsafeBytes {
                          ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/Message+BinaryAdditions.swift:36:14: warning: 'withUnsafeMutableBytes' is deprecated: use `withUnsafeMutableBytes<R>(_: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R` instead
    try data.withUnsafeMutableBytes { (pointer: UnsafeMutablePointer<UInt8>) in
             ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/Message+BinaryAdditions.swift:107:16: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
      try data.withUnsafeBytes { (pointer: UnsafePointer<UInt8>) in
               ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/Message+JSONAdditions.swift:92:22: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
    try jsonUTF8Data.withUnsafeBytes { (bytes:UnsafePointer<UInt8>) in
                     ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/Message+JSONArrayAdditions.swift:94:29: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
    return try jsonUTF8Data.withUnsafeBytes { (bytes:UnsafePointer<UInt8>) in
                            ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/Message+TextFormatAdditions.swift:56:18: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
        try data.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) in
                 ^
tProtobuf/TextFormatEncoder.swift:266:15: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
        value.withUnsafeBytes { (p: UnsafePointer<UInt8>) in
              ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/AnyMessageStorage.swift:54:16: warning: 'init(bytes:)' is deprecated: use `init(_:)` instead
  var result = Data(bytes: [asciiOpenCurlyBracket])
               ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/AnyMessageStorage.swift:69:19: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
  try contentJSON.withUnsafeBytes { (bytes:UnsafePointer<UInt8>) in
                  ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/BinaryDecoder.swift:879:23: warning: 'withUnsafeMutableBytes' is deprecated: use `withUnsafeMutableBytes<R>(_: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R` instead
                field.withUnsafeMutableBytes { (pointer: UnsafeMutablePointer<UInt8>) in
                      ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/BinaryDecoder.swift:1207:30: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
                    try data.withUnsafeBytes { (pointer: UnsafePointer<UInt8>) in
                             ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/BinaryDecoder.swift:1248:33: warning: 'withUnsafeMutableBytes' is deprecated: use `withUnsafeMutableBytes<R>(_: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R` instead
                        payload.withUnsafeMutableBytes { (pointer: UnsafeMutablePointer<UInt8>) in
                                ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/BinaryDelimited.swift:58:10: warning: 'withUnsafeMutableBytes' is deprecated: use `withUnsafeMutableBytes<R>(_: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R` instead
    data.withUnsafeMutableBytes { (pointer: UnsafeMutablePointer<UInt8>) in
         ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/BinaryDelimited.swift:64:10: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
    data.withUnsafeBytes { (pointer: UnsafePointer<UInt8>) in
         ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/BinaryDelimited.swift:157:10: warning: 'withUnsafeMutableBytes' is deprecated: use `withUnsafeMutableBytes<R>(_: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R` instead
    data.withUnsafeMutableBytes { (pointer: UnsafeMutablePointer<UInt8>) in
         ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/Google_Protobuf_Any+Extensions.swift:69:18: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
        try data.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) in
                 ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/JSONEncoder.swift:73:44: warning: 'init(bytes:)' is deprecated: use `init(_:)` instead
    internal var dataResult: Data { return Data(bytes: data) }
                                           ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/JSONEncoder.swift:336:19: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
            value.withUnsafeBytes { (p: UnsafePointer<UInt8>) in
                  ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/TextFormatEncodingVisitor.swift:71:17: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
      try bytes.withUnsafeBytes { (p: UnsafePointer<UInt8>) -> () in
                ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/TextFormatEncodingVisitor.swift:100:21: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
              bytes.withUnsafeBytes { (p: UnsafePointer<UInt8>) -> () in
                    ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/SwiftProtobuf/TextFormatScanner.swift:358:12: warning: 'withUnsafeMutableBytes' is deprecated: use `withUnsafeMutableBytes<R>(_: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R` instead
      data.withUnsafeMutableBytes {
           ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/protoc-gen-swift/FileIo.swift:50:11: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
    bytes.withUnsafeBytes { (p: UnsafePointer<UInt8>) -> () in
          ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/.build/checkouts/swift-protobuf/Sources/protoc-gen-swift/FileIo.swift:71:16: warning: 'init(bytes:)' is deprecated: use `init(_:)` instead
        return Data(bytes: buff)
               ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/Sources/protoc-gen-swiftgrpc/io.swift:49:16: warning: 'init(bytes:)' is deprecated: use `init(_:)` instead
        return Data(bytes: buff)
               ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/Sources/protoc-gen-swiftgrpc/io.swift:58:11: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
    bytes.withUnsafeBytes { (p: UnsafePointer<UInt8>) -> Void in
          ^
/Users/johannes/.Trash/.nio-release-tools_WPakBo/grpc-swift/Sources/SwiftGRPC/Core/ByteBuffer.swift:39:10: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
    data.withUnsafeBytes { bytes in
         ^
[1138/1138] Compiling BoringSSL crypto/asn1/a_bitstr.c
Test Suite 'All tests' started at 2019-04-17 09:57:41.800
Test Suite 'debug.xctest' started at 2019-04-17 09:57:41.825
Test Suite 'ClientThrowingWhenServerReturningErrorTests' started at 2019-04-17 09:57:41.825
Test Case 'ClientThrowingWhenServerReturningErrorTests.testBidirectionalStreaming' started at 2019-04-17 09:57:41.825

[stuck]
```

--- a/Sources/SwiftGRPCNIO/GRPCClientConnection.swift
+++ b/Sources/SwiftGRPCNIO/GRPCClientConnection.swift
@@ -39,24 +39,22 @@ open class GRPCClientConnection {
     tls tlsMode: TLSMode = .none,
     hostOverride: String? = nil
   ) throws -> EventLoopFuture<GRPCClientConnection> {
-    // We need to capture the multiplexer from the channel initializer to store it after connection.
-    let multiplexerPromise: EventLoopPromise<HTTP2StreamMultiplexer> = eventLoopGroup.next().makePromise()
-
     let bootstrap = ClientBootstrap(group: eventLoopGroup)
       // Enable SO_REUSEADDR.
       .channelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
       .channelInitializer { channel in
-        let multiplexer = configureTLS(mode: tlsMode, channel: channel, host: hostOverride ?? host).flatMap {
+        configureTLS(mode: tlsMode, channel: channel, host: hostOverride ?? host).flatMap {
           channel.configureHTTP2Pipeline(mode: .client)
-        }
-
-        multiplexer.cascade(to: multiplexerPromise)
-        return multiplexer.map { _ in }
+        }.map { _ in }
       }
 
-    return bootstrap.connect(host: host, port: port)
-      .and(multiplexerPromise.futureResult)
-      .map { channel, multiplexer in GRPCClientConnection(channel: channel, multiplexer: multiplexer, host: host, httpProtocol: tlsMode.httpProtocol) }
+    return bootstrap.connect(host: host, port: port).flatMap { channel in
+      channel.pipeline.context(handlerType: HTTP2StreamMultiplexer.self).map { context in
+        context.handler as! HTTP2StreamMultiplexer
+      }.map { multiplexer in
+        GRPCClientConnection(channel: channel, multiplexer: multiplexer, host: host, httpProtocol: tlsMode.httpProtocol)
+      }
+    }
   }
 
   /// Configure an SSL handler on the channel, if one is required.

